src/expr.rs

```diff
@@ -674,3 +700,57 @@ fn rewrite_unary_op(context: &RewriteContext,
 
     Some(format!("{}{}", operator_str, subexpr))
 }
+
+fn rewrite_assign(context: &RewriteContext,
+                  lhs: &ast::Expr,
+                  rhs: &ast::Expr,
+                  op: Option<&ast::BinOp>,
+                  width: usize,
+                  offset: usize)
+                  -> Option<String> {
+    let operator_str = match op {
+        Some(op) => context.codemap.span_to_snippet(op.span).unwrap(),
+        None => "=".to_owned()
+    };
+
+    // 2 = " =".len()
+    let lhs_str = format!("{} {}",
+                          try_opt!(lhs.rewrite(context, width - operator_str.len() - 2, offset)),
+                          operator_str);
+
+    rewrite_assignment(&context, lhs_str, rhs, width, offset)
+}
+
+// The left hand side must contain everything up to, and including, the
+// assignment operator.
+pub fn rewrite_assignment<S: Into<String>>(context: &RewriteContext,
```

Comment:
> Could we factor this differently? I find it a little odd atm because it doesn't fit the hierarchical pattern which most reformatting does, i.e., we're not rewriting a sub-expression, we delegating responsibility for formatting the whole expression to the rewrite function for a sub-expression.

> I see that you want to share code between assignment and let, which seems sensible. And this kind of fits with a kind of inline formatting scheme - i.e., you make line splitting decisions as you go using only info from previous expressions, rather than from all sub-expressions. Is this deliberate? Does it work better than computing the rhs string first and then stitching them together?

-nrc